File: aidp-java-client/src/main/java/com/oracle/aidataplatform/dp/model/KnowledgeBaseSummary.java
package com.oracle.aidataplatform.dp.model;



/**
 * Summary of the KnowledgeBase.
**/
@jakarta.annotation.Generated(value = "OracleSDKGenerator", comments = "API Version: 20260430")
@com.fasterxml.jackson.databind.annotation.JsonDeserialize(builder=KnowledgeBaseSummary.Builder.class)

public final class KnowledgeBaseSummary  {
    @Deprecated
    @java.beans.ConstructorProperties({"key", "displayName", "catalogKey", "schemaKey", "description", "timeCreated", "createdBy", "timeUpdated", "updatedBy", "currentProcessedItems", "lifecycleState", "systemTags"})
    public KnowledgeBaseSummary(String key, String displayName, String catalogKey, String schemaKey, String description, java.util.Date timeCreated, String createdBy, java.util.Date timeUpdated, String updatedBy, Long currentProcessedItems, KnowledgeBaseLifecycleState lifecycleState, java.util.Map<String, java.util.Map<String, Object>> systemTags) {
        super();
        this.key = key;
        this.displayName = displayName;
        this.catalogKey = catalogKey;
        this.schemaKey = schemaKey;
        this.description = description;
        this.timeCreated = timeCreated;
        this.createdBy = createdBy;
        this.timeUpdated = timeUpdated;
        this.updatedBy = updatedBy;
        this.currentProcessedItems = currentProcessedItems;
        this.lifecycleState = lifecycleState;
        this.systemTags = systemTags;
    }

    @com.fasterxml.jackson.databind.annotation.JsonPOJOBuilder(withPrefix = "")
    public static class Builder {
                /**
     * Immutable Unique identifier generated at creation
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("key")
private String key;

        /**
         * Immutable Unique identifier generated at creation
         * @param key the value to set
         * @return this builder
         **/
        

public Builder key(String key) {
    this.key = key;
    return this;
}
            /**
     * KnowledgeBase Identifier, can be renamed
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("displayName")
private String displayName;

        /**
         * KnowledgeBase Identifier, can be renamed
         * @param displayName the value to set
         * @return this builder
         **/
        

public Builder displayName(String displayName) {
    this.displayName = displayName;
    return this;
}
            /**
     * the catalog hosting the KnowledgeBase
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("catalogKey")
private String catalogKey;

        /**
         * the catalog hosting the KnowledgeBase
         * @param catalogKey the value to set
         * @return this builder
         **/
        

public Builder catalogKey(String catalogKey) {
    this.catalogKey = catalogKey;
    return this;
}
            /**
     * the schema inside the catalog hosting the KnowledgeBase
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("schemaKey")
private String schemaKey;

        /**
         * the schema inside the catalog hosting the KnowledgeBase
         * @param schemaKey the value to set
         * @return this builder
         **/
        

public Builder schemaKey(String schemaKey) {
    this.schemaKey = schemaKey;
    return this;
}
            /**
     * The description of KnowledgeBase.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("description")
private String description;

        /**
         * The description of KnowledgeBase.
         * @param description the value to set
         * @return this builder
         **/
        

public Builder description(String description) {
    this.description = description;
    return this;
}
            /**
     * The time at which KnowledgeBase was created. An RFC3339 formatted datetime string
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
private java.util.Date timeCreated;

        /**
         * The time at which KnowledgeBase was created. An RFC3339 formatted datetime string
         * @param timeCreated the value to set
         * @return this builder
         **/
        

public Builder timeCreated(java.util.Date timeCreated) {
    this.timeCreated = timeCreated;
    return this;
}
            /**
     * Identifier for KnowledgeBase creator
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("createdBy")
private String createdBy;

        /**
         * Identifier for KnowledgeBase creator
         * @param createdBy the value to set
         * @return this builder
         **/
        

public Builder createdBy(String createdBy) {
    this.createdBy = createdBy;
    return this;
}
            /**
     * The time at which KnowledgeBase was updated. An RFC3339 formatted datetime string
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
private java.util.Date timeUpdated;

        /**
         * The time at which KnowledgeBase was updated. An RFC3339 formatted datetime string
         * @param timeUpdated the value to set
         * @return this builder
         **/
        

public Builder timeUpdated(java.util.Date timeUpdated) {
    this.timeUpdated = timeUpdated;
    return this;
}
            /**
     * Identifier for principal who updated the KnowledgeBase
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("updatedBy")
private String updatedBy;

        /**
         * Identifier for principal who updated the KnowledgeBase
         * @param updatedBy the value to set
         * @return this builder
         **/
        

public Builder updatedBy(String updatedBy) {
    this.updatedBy = updatedBy;
    return this;
}
            /**
     * Count of items/documents processed by KB for which embeddings are present
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("currentProcessedItems")
private Long currentProcessedItems;

        /**
         * Count of items/documents processed by KB for which embeddings are present
         * @param currentProcessedItems the value to set
         * @return this builder
         **/
        

public Builder currentProcessedItems(Long currentProcessedItems) {
    this.currentProcessedItems = currentProcessedItems;
    return this;
}
            /**
     * The current state of the KnowledgeBase.
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("lifecycleState")
private KnowledgeBaseLifecycleState lifecycleState;

        /**
         * The current state of the KnowledgeBase.
         * @param lifecycleState the value to set
         * @return this builder
         **/
        

public Builder lifecycleState(KnowledgeBaseLifecycleState lifecycleState) {
    this.lifecycleState = lifecycleState;
    return this;
}
            /**
     * System tags for this resource. Each key is predefined and scoped to a namespace.
* <p>
Example: {@code {"orcl-cloud": {"free-tier-retained": "true"}}}
* 
     **/
    
@com.fasterxml.jackson.annotation.JsonProperty("systemTags")
private java.util.Map<String, java.util.Map<String, Object>> systemTags;

        /**
         * System tags for this resource. Each key is predefined and scoped to a namespace.
* <p>
Example: {@code {"orcl-cloud": {"free-tier-retained": "true"}}}
* 
         * @param systemTags the value to set
         * @return this builder
         **/
        

public Builder systemTags(java.util.Map<String, java.util.Map<String, Object>> systemTags) {
    this.systemTags = systemTags;
    return this;
}


        public KnowledgeBaseSummary build() {
            KnowledgeBaseSummary model = new KnowledgeBaseSummary(this.key
                , this.displayName
                , this.catalogKey
                , this.schemaKey
                , this.description
                , this.timeCreated
                , this.createdBy
                , this.timeUpdated
                , this.updatedBy
                , this.currentProcessedItems
                , this.lifecycleState
                , this.systemTags);            return model;
        }

        @com.fasterxml.jackson.annotation.JsonIgnore
        public Builder copy(KnowledgeBaseSummary model) {
                this.key(model.getKey());
    this.displayName(model.getDisplayName());
    this.catalogKey(model.getCatalogKey());
    this.schemaKey(model.getSchemaKey());
    this.description(model.getDescription());
    this.timeCreated(model.getTimeCreated());
    this.createdBy(model.getCreatedBy());
    this.timeUpdated(model.getTimeUpdated());
    this.updatedBy(model.getUpdatedBy());
    this.currentProcessedItems(model.getCurrentProcessedItems());
    this.lifecycleState(model.getLifecycleState());
    this.systemTags(model.getSystemTags());
return this;
        }
    }

    /**
     * Create a new builder.
     */
    public static Builder builder() {
        return new Builder();
    }


    public Builder toBuilder() {
        return new Builder().copy(this);
    }

    


        /**
     * Immutable Unique identifier generated at creation
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("key")
    private final String key;

        /**
     * Immutable Unique identifier generated at creation
     * @return the value
     **/
    
    public String getKey() {
        return key;
    }


        /**
     * KnowledgeBase Identifier, can be renamed
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("displayName")
    private final String displayName;

        /**
     * KnowledgeBase Identifier, can be renamed
     * @return the value
     **/
    
    public String getDisplayName() {
        return displayName;
    }


        /**
     * the catalog hosting the KnowledgeBase
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("catalogKey")
    private final String catalogKey;

        /**
     * the catalog hosting the KnowledgeBase
     * @return the value
     **/
    
    public String getCatalogKey() {
        return catalogKey;
    }


        /**
     * the schema inside the catalog hosting the KnowledgeBase
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("schemaKey")
    private final String schemaKey;

        /**
     * the schema inside the catalog hosting the KnowledgeBase
     * @return the value
     **/
    
    public String getSchemaKey() {
        return schemaKey;
    }


        /**
     * The description of KnowledgeBase.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("description")
    private final String description;

        /**
     * The description of KnowledgeBase.
     * @return the value
     **/
    
    public String getDescription() {
        return description;
    }


        /**
     * The time at which KnowledgeBase was created. An RFC3339 formatted datetime string
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeCreated")
    private final java.util.Date timeCreated;

        /**
     * The time at which KnowledgeBase was created. An RFC3339 formatted datetime string
     * @return the value
     **/
    
    public java.util.Date getTimeCreated() {
        return timeCreated;
    }


        /**
     * Identifier for KnowledgeBase creator
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("createdBy")
    private final String createdBy;

        /**
     * Identifier for KnowledgeBase creator
     * @return the value
     **/
    
    public String getCreatedBy() {
        return createdBy;
    }


        /**
     * The time at which KnowledgeBase was updated. An RFC3339 formatted datetime string
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("timeUpdated")
    private final java.util.Date timeUpdated;

        /**
     * The time at which KnowledgeBase was updated. An RFC3339 formatted datetime string
     * @return the value
     **/
    
    public java.util.Date getTimeUpdated() {
        return timeUpdated;
    }


        /**
     * Identifier for principal who updated the KnowledgeBase
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("updatedBy")
    private final String updatedBy;

        /**
     * Identifier for principal who updated the KnowledgeBase
     * @return the value
     **/
    
    public String getUpdatedBy() {
        return updatedBy;
    }


        /**
     * Count of items/documents processed by KB for which embeddings are present
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("currentProcessedItems")
    private final Long currentProcessedItems;

        /**
     * Count of items/documents processed by KB for which embeddings are present
     * @return the value
     **/
    
    public Long getCurrentProcessedItems() {
        return currentProcessedItems;
    }

    
        /**
     * The current state of the KnowledgeBase.
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("lifecycleState")
    private final KnowledgeBaseLifecycleState lifecycleState;

        /**
     * The current state of the KnowledgeBase.
     * @return the value
     **/
    
    public KnowledgeBaseLifecycleState getLifecycleState() {
        return lifecycleState;
    }


        /**
     * System tags for this resource. Each key is predefined and scoped to a namespace.
* <p>
Example: {@code {"orcl-cloud": {"free-tier-retained": "true"}}}
* 
     **/
    
    @com.fasterxml.jackson.annotation.JsonProperty("systemTags")
    private final java.util.Map<String, java.util.Map<String, Object>> systemTags;

        /**
     * System tags for this resource. Each key is predefined and scoped to a namespace.
* <p>
Example: {@code {"orcl-cloud": {"free-tier-retained": "true"}}}
* 
     * @return the value
     **/
    
    public java.util.Map<String, java.util.Map<String, Object>> getSystemTags() {
        return systemTags;
    }

    @Override
    public String toString() {
        return this.toString(true);
    }

    /**
     * Return a string representation of the object.
     * @param includeByteArrayContents true to include the full contents of byte arrays
     * @return string representation
     */
    public String toString(boolean includeByteArrayContents) {
        java.lang.StringBuilder sb = new java.lang.StringBuilder();
        sb.append("KnowledgeBaseSummary(");
        sb.append("key=").append(String.valueOf(this.key));
        sb.append(", displayName=").append(String.valueOf(this.displayName));
        sb.append(", catalogKey=").append(String.valueOf(this.catalogKey));
        sb.append(", schemaKey=").append(String.valueOf(this.schemaKey));
        sb.append(", description=").append(String.valueOf(this.description));
        sb.append(", timeCreated=").append(String.valueOf(this.timeCreated));
        sb.append(", createdBy=").append(String.valueOf(this.createdBy));
        sb.append(", timeUpdated=").append(String.valueOf(this.timeUpdated));
        sb.append(", updatedBy=").append(String.valueOf(this.updatedBy));
        sb.append(", currentProcessedItems=").append(String.valueOf(this.currentProcessedItems));
        sb.append(", lifecycleState=").append(String.valueOf(this.lifecycleState));
        sb.append(", systemTags=").append(String.valueOf(this.systemTags));
        sb.append(")");
        return sb.toString();
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) {
            return true;
        }
        if (!(o instanceof KnowledgeBaseSummary)) {
            return false;
        }

        KnowledgeBaseSummary other = (KnowledgeBaseSummary) o;
        return java.util.Objects.equals(this.key, other.key) &&
            java.util.Objects.equals(this.displayName, other.displayName) &&
            java.util.Objects.equals(this.catalogKey, other.catalogKey) &&
            java.util.Objects.equals(this.schemaKey, other.schemaKey) &&
            java.util.Objects.equals(this.description, other.description) &&
            java.util.Objects.equals(this.timeCreated, other.timeCreated) &&
            java.util.Objects.equals(this.createdBy, other.createdBy) &&
            java.util.Objects.equals(this.timeUpdated, other.timeUpdated) &&
            java.util.Objects.equals(this.updatedBy, other.updatedBy) &&
            java.util.Objects.equals(this.currentProcessedItems, other.currentProcessedItems) &&
            java.util.Objects.equals(this.lifecycleState, other.lifecycleState) &&
            java.util.Objects.equals(this.systemTags, other.systemTags);
    }

    @Override
    public int hashCode() {
        final int PRIME = 59;
        int result = 1;
        result = (result * PRIME) + (this.key == null ? 43 : this.key.hashCode());
        result = (result * PRIME) + (this.displayName == null ? 43 : this.displayName.hashCode());
        result = (result * PRIME) + (this.catalogKey == null ? 43 : this.catalogKey.hashCode());
        result = (result * PRIME) + (this.schemaKey == null ? 43 : this.schemaKey.hashCode());
        result = (result * PRIME) + (this.description == null ? 43 : this.description.hashCode());
        result = (result * PRIME) + (this.timeCreated == null ? 43 : this.timeCreated.hashCode());
        result = (result * PRIME) + (this.createdBy == null ? 43 : this.createdBy.hashCode());
        result = (result * PRIME) + (this.timeUpdated == null ? 43 : this.timeUpdated.hashCode());
        result = (result * PRIME) + (this.updatedBy == null ? 43 : this.updatedBy.hashCode());
        result = (result * PRIME) + (this.currentProcessedItems == null ? 43 : this.currentProcessedItems.hashCode());
        result = (result * PRIME) + (this.lifecycleState == null ? 43 : this.lifecycleState.hashCode());
        result = (result * PRIME) + (this.systemTags == null ? 43 : this.systemTags.hashCode());
        return result;
    }


}

File: aidp-python-client/src/aidp_python_client/aidataplatform_dp/models/knowledge_base_summary.py
# coding: utf-8
# Copyright (c) 2026, Oracle and/or its affiliates.  All rights reserved.



from oci.util import formatted_flat_dict, NONE_SENTINEL, value_allowed_none_or_none_sentinel  # noqa: F401
from oci.decorators import init_model_state_from_kwargs


@init_model_state_from_kwargs
class KnowledgeBaseSummary(object):
    """
    Summary of the KnowledgeBase.
    """

    #: A constant which can be used with the lifecycle_state property of a KnowledgeBaseSummary.
    #: This constant has a value of "CREATING"
    LIFECYCLE_STATE_CREATING = "CREATING"

    #: A constant which can be used with the lifecycle_state property of a KnowledgeBaseSummary.
    #: This constant has a value of "ACTIVE"
    LIFECYCLE_STATE_ACTIVE = "ACTIVE"

    #: A constant which can be used with the lifecycle_state property of a KnowledgeBaseSummary.
    #: This constant has a value of "UPDATING"
    LIFECYCLE_STATE_UPDATING = "UPDATING"

    #: A constant which can be used with the lifecycle_state property of a KnowledgeBaseSummary.
    #: This constant has a value of "DELETING"
    LIFECYCLE_STATE_DELETING = "DELETING"

    #: A constant which can be used with the lifecycle_state property of a KnowledgeBaseSummary.
    #: This constant has a value of "NEEDS_ATTENTION"
    LIFECYCLE_STATE_NEEDS_ATTENTION = "NEEDS_ATTENTION"

    #: A constant which can be used with the lifecycle_state property of a KnowledgeBaseSummary.
    #: This constant has a value of "DELETED"
    LIFECYCLE_STATE_DELETED = "DELETED"

    #: A constant which can be used with the lifecycle_state property of a KnowledgeBaseSummary.
    #: This constant has a value of "FAILED"
    LIFECYCLE_STATE_FAILED = "FAILED"

    def __init__(self, **kwargs):
        """
        Initializes a new KnowledgeBaseSummary object with values from keyword arguments.
        The following keyword arguments are supported (corresponding to the getters/setters of this class):

        :param key:
            The value to assign to the key property of this KnowledgeBaseSummary.
        :type key: str

        :param display_name:
            The value to assign to the display_name property of this KnowledgeBaseSummary.
        :type display_name: str

        :param catalog_key:
            The value to assign to the catalog_key property of this KnowledgeBaseSummary.
        :type catalog_key: str

        :param schema_key:
            The value to assign to the schema_key property of this KnowledgeBaseSummary.
        :type schema_key: str

        :param description:
            The value to assign to the description property of this KnowledgeBaseSummary.
        :type description: str

        :param time_created:
            The value to assign to the time_created property of this KnowledgeBaseSummary.
        :type time_created: datetime

        :param created_by:
            The value to assign to the created_by property of this KnowledgeBaseSummary.
        :type created_by: str

        :param time_updated:
            The value to assign to the time_updated property of this KnowledgeBaseSummary.
        :type time_updated: datetime

        :param updated_by:
            The value to assign to the updated_by property of this KnowledgeBaseSummary.
        :type updated_by: str

        :param current_processed_items:
            The value to assign to the current_processed_items property of this KnowledgeBaseSummary.
        :type current_processed_items: int

        :param lifecycle_state:
            The value to assign to the lifecycle_state property of this KnowledgeBaseSummary.
            Allowed values for this property are: "CREATING", "ACTIVE", "UPDATING", "DELETING", "NEEDS_ATTENTION", "DELETED", "FAILED"
        :type lifecycle_state: str

        :param system_tags:
            The value to assign to the system_tags property of this KnowledgeBaseSummary.
        :type system_tags: dict(str, dict(str, object))

        """
        self.swagger_types = {
            'key': 'str',
            'display_name': 'str',
            'catalog_key': 'str',
            'schema_key': 'str',
            'description': 'str',
            'time_created': 'datetime',
            'created_by': 'str',
            'time_updated': 'datetime',
            'updated_by': 'str',
            'current_processed_items': 'int',
            'lifecycle_state': 'str',
            'system_tags': 'dict(str, dict(str, object))'
        }

        self.attribute_map = {
            'key': 'key',
            'display_name': 'displayName',
            'catalog_key': 'catalogKey',
            'schema_key': 'schemaKey',
            'description': 'description',
            'time_created': 'timeCreated',
            'created_by': 'createdBy',
            'time_updated': 'timeUpdated',
            'updated_by': 'updatedBy',
            'current_processed_items': 'currentProcessedItems',
            'lifecycle_state': 'lifecycleState',
            'system_tags': 'systemTags'
        }

        self._key = None
        self._display_name = None
        self._catalog_key = None
        self._schema_key = None
        self._description = None
        self._time_created = None
        self._created_by = None
        self._time_updated = None
        self._updated_by = None
        self._current_processed_items = None
        self._lifecycle_state = None
        self._system_tags = None

    @property
    def key(self):
        """
        **[Required]** Gets the key of this KnowledgeBaseSummary.
        Immutable Unique identifier generated at creation


        :return: The key of this KnowledgeBaseSummary.
        :rtype: str
        """
        return self._key

    @key.setter
    def key(self, key):
        """
        Sets the key of this KnowledgeBaseSummary.
        Immutable Unique identifier generated at creation


        :param key: The key of this KnowledgeBaseSummary.
        :type: str
        """
        self._key = key

    @property
    def display_name(self):
        """
        **[Required]** Gets the display_name of this KnowledgeBaseSummary.
        KnowledgeBase Identifier, can be renamed


        :return: The display_name of this KnowledgeBaseSummary.
        :rtype: str
        """
        return self._display_name

    @display_name.setter
    def display_name(self, display_name):
        """
        Sets the display_name of this KnowledgeBaseSummary.
        KnowledgeBase Identifier, can be renamed


        :param display_name: The display_name of this KnowledgeBaseSummary.
        :type: str
        """
        self._display_name = display_name

    @property
    def catalog_key(self):
        """
        Gets the catalog_key of this KnowledgeBaseSummary.
        the catalog hosting the KnowledgeBase


        :return: The catalog_key of this KnowledgeBaseSummary.
        :rtype: str
        """
        return self._catalog_key

    @catalog_key.setter
    def catalog_key(self, catalog_key):
        """
        Sets the catalog_key of this KnowledgeBaseSummary.
        the catalog hosting the KnowledgeBase


        :param catalog_key: The catalog_key of this KnowledgeBaseSummary.
        :type: str
        """
        self._catalog_key = catalog_key

    @property
    def schema_key(self):
        """
        Gets the schema_key of this KnowledgeBaseSummary.
        the schema inside the catalog hosting the KnowledgeBase


        :return: The schema_key of this KnowledgeBaseSummary.
        :rtype: str
        """
        return self._schema_key

    @schema_key.setter
    def schema_key(self, schema_key):
        """
        Sets the schema_key of this KnowledgeBaseSummary.
        the schema inside the catalog hosting the KnowledgeBase


        :param schema_key: The schema_key of this KnowledgeBaseSummary.
        :type: str
        """
        self._schema_key = schema_key

    @property
    def description(self):
        """
        Gets the description of this KnowledgeBaseSummary.
        The description of KnowledgeBase.


        :return: The description of this KnowledgeBaseSummary.
        :rtype: str
        """
        return self._description

    @description.setter
    def description(self, description):
        """
        Sets the description of this KnowledgeBaseSummary.
        The description of KnowledgeBase.


        :param description: The description of this KnowledgeBaseSummary.
        :type: str
        """
        self._description = description

    @property
    def time_created(self):
        """
        **[Required]** Gets the time_created of this KnowledgeBaseSummary.
        The time at which KnowledgeBase was created. An RFC3339 formatted datetime string


        :return: The time_created of this KnowledgeBaseSummary.
        :rtype: datetime
        """
        return self._time_created

    @time_created.setter
    def time_created(self, time_created):
        """
        Sets the time_created of this KnowledgeBaseSummary.
        The time at which KnowledgeBase was created. An RFC3339 formatted datetime string


        :param time_created: The time_created of this KnowledgeBaseSummary.
        :type: datetime
        """
        self._time_created = time_created

    @property
    def created_by(self):
        """
        Gets the created_by of this KnowledgeBaseSummary.
        Identifier for KnowledgeBase creator


        :return: The created_by of this KnowledgeBaseSummary.
        :rtype: str
        """
        return self._created_by

    @created_by.setter
    def created_by(self, created_by):
        """
        Sets the created_by of this KnowledgeBaseSummary.
        Identifier for KnowledgeBase creator


        :param created_by: The created_by of this KnowledgeBaseSummary.
        :type: str
        """
        self._created_by = created_by

    @property
    def time_updated(self):
        """
        Gets the time_updated of this KnowledgeBaseSummary.
        The time at which KnowledgeBase was updated. An RFC3339 formatted datetime string


        :return: The time_updated of this KnowledgeBaseSummary.
        :rtype: datetime
        """
        return self._time_updated

    @time_updated.setter
    def time_updated(self, time_updated):
        """
        Sets the time_updated of this KnowledgeBaseSummary.
        The time at which KnowledgeBase was updated. An RFC3339 formatted datetime string


        :param time_updated: The time_updated of this KnowledgeBaseSummary.
        :type: datetime
        """
        self._time_updated = time_updated

    @property
    def updated_by(self):
        """
        Gets the updated_by of this KnowledgeBaseSummary.
        Identifier for principal who updated the KnowledgeBase


        :return: The updated_by of this KnowledgeBaseSummary.
        :rtype: str
        """
        return self._updated_by

    @updated_by.setter
    def updated_by(self, updated_by):
        """
        Sets the updated_by of this KnowledgeBaseSummary.
        Identifier for principal who updated the KnowledgeBase


        :param updated_by: The updated_by of this KnowledgeBaseSummary.
        :type: str
        """
        self._updated_by = updated_by

    @property
    def current_processed_items(self):
        """
        Gets the current_processed_items of this KnowledgeBaseSummary.
        Count of items/documents processed by KB for which embeddings are present


        :return: The current_processed_items of this KnowledgeBaseSummary.
        :rtype: int
        """
        return self._current_processed_items

    @current_processed_items.setter
    def current_processed_items(self, current_processed_items):
        """
        Sets the current_processed_items of this KnowledgeBaseSummary.
        Count of items/documents processed by KB for which embeddings are present


        :param current_processed_items: The current_processed_items of this KnowledgeBaseSummary.
        :type: int
        """
        self._current_processed_items = current_processed_items

    @property
    def lifecycle_state(self):
        """
        **[Required]** Gets the lifecycle_state of this KnowledgeBaseSummary.
        The current state of the KnowledgeBase.

        Allowed values for this property are: "CREATING", "ACTIVE", "UPDATING", "DELETING", "NEEDS_ATTENTION", "DELETED", "FAILED"


        :return: The lifecycle_state of this KnowledgeBaseSummary.
        :rtype: str
        """
        return self._lifecycle_state

    @lifecycle_state.setter
    def lifecycle_state(self, lifecycle_state):
        """
        Sets the lifecycle_state of this KnowledgeBaseSummary.
        The current state of the KnowledgeBase.


        :param lifecycle_state: The lifecycle_state of this KnowledgeBaseSummary.
        :type: str
        """
        allowed_values = ["CREATING", "ACTIVE", "UPDATING", "DELETING", "NEEDS_ATTENTION", "DELETED", "FAILED"]
        if not value_allowed_none_or_none_sentinel(lifecycle_state, allowed_values):
            raise ValueError(
                "Invalid value for `lifecycle_state`, must be None or one of {0}"
                .format(allowed_values)
            )
        self._lifecycle_state = lifecycle_state

    @property
    def system_tags(self):
        """
        Gets the system_tags of this KnowledgeBaseSummary.
        System tags for this resource. Each key is predefined and scoped to a namespace.

        Example: `{\"orcl-cloud\": {\"free-tier-retained\": \"true\"}}`


        :return: The system_tags of this KnowledgeBaseSummary.
        :rtype: dict(str, dict(str, object))
        """
        return self._system_tags

    @system_tags.setter
    def system_tags(self, system_tags):
        """
        Sets the system_tags of this KnowledgeBaseSummary.
        System tags for this resource. Each key is predefined and scoped to a namespace.

        Example: `{\"orcl-cloud\": {\"free-tier-retained\": \"true\"}}`


        :param system_tags: The system_tags of this KnowledgeBaseSummary.
        :type: dict(str, dict(str, object))
        """
        self._system_tags = system_tags

    def __repr__(self):
        return formatted_flat_dict(self)

    def __eq__(self, other):
        if other is None:
            return False

        return self.__dict__ == other.__dict__

    def __ne__(self, other):
        return not self == other
